File: tests/CompileTests/OpenMP_tests/fortran/task_largenumber.f
******************************************      
* Simplest OMP task example
* Liao 1/21/2010
******************************************
      program main
      include 'omp_lib.h'
      integer i
      integer item(10)
      external process
      do i = 1, 10
        item(i) = 11-i
      enddo  

!$omp parallel
!$omp single
      print *, 'using ', omp_get_num_threads(), ' threads' 
      do i = 1, 10
!$omp task 
      call process(i,item)
!$omp end task
      continue ! use a no-op stmt to get comment position right, a workaround
      enddo  
!$omp end single
!$omp end parallel
      end 

******************************************
      subroutine process(input,item)
      include 'omp_lib.h'
      integer input
      integer item(10)
      print *, 'idx ', input,'-> ', item(input)
      print *, 'by thread ', omp_get_thread_num()
      end

File: tests/nonsmoke/functional/CompileTests/OpenMP_tests/task_largenumber.c
#include <stdio.h>
#include <omp.h>

#define LARGE_NUMBER 10
//#define LARGE_NUMBER 10000000
double item[LARGE_NUMBER];
void process (double input)
{
  printf("processing %f by thread %d\n",input, omp_get_thread_num());
}
int cutoff =1000;
int
main ()
{
#pragma omp parallel
  {
#pragma omp single
    {
      int i;
      printf("Using %d threads.\n",omp_get_num_threads());
      for (i = 0; i < LARGE_NUMBER; i++)
        #pragma omp task if (i < cutoff)
         process (item[i]);
    }
  }
  return 0;
}
